### PR TITLE
 adapt java-client: support cpyher query api

### PR DESCRIPTION
**Pull request by jadepeng**
adapt java-client for hugegraph (see https://github.com/apache/incubator-hugegraph/pull/1866)

### CODE REVIEW

File: hugegraph-client/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    protected String type() {
+        return HugeType.CYPHER.string();
+    }
+
+    public Response post(String cypher) {
+        RestResult result = this.client.post(this.path(), cypher);
+        return result.readObject(Response.class);
+    }
+}
```
Comment (javeme) [inline hugegraph-client/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:46]: expect a blank line at the end of file

**Comment by javeme** [comment]
The code looks fine, wait for the server https://github.com/apache/incubator-hugegraph/pull/1866 PR merged~
@jadepeng Did you test it locally?

**Comment by jadepeng** [comment]
> The code looks fine, wait for the server [apache/incubator-hugegraph#1866](https://github.com/apache/incubator-hugegraph/pull/1866) PR merged~ @jadepeng Did you test it locally?

Tested

**Comment by javeme** [comment]
now we can update the latest commit-id and re-run ci: https://github.com/apache/incubator-hugegraph-toolchain/blob/master/.github/workflows/client-ci.yml#L26

File: hugegraph-client/src/main/java/com/baidu/hugegraph/structure/constant/HugeType.java [hunk 2]
```diff
@@ -60,7 +60,10 @@ public enum HugeType {
     VERSION(230, "versions"),
 
     // Metrics
-    METRICS(240, "metrics");
+    METRICS(240, "metrics"),
+
+    // Cypher
+    CYPHER(250, "cypher");
```
Comment (javeme) [inline hugegraph-client/src/main/java/com/baidu/hugegraph/structure/constant/HugeType.java:66]: can move to line 56 and set code=202

**Review by javeme — APPROVED** [review]

File: hugegraph-client/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 HugeGraph Authors
```
Comment (imbajin) [inline hugegraph-client/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:2]: better to use 2022?

File: hugegraph-client/src/main/java/com/baidu/hugegraph/driver/CypherManager.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 HugeGraph Authors
```
Comment (imbajin) [inline hugegraph-client/src/main/java/com/baidu/hugegraph/driver/CypherManager.java:2]: same

File: hugegraph-client/src/test/java/com/baidu/hugegraph/api/CypherApiTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 HugeGraph Authors
```
Comment (imbajin) [inline hugegraph-client/src/test/java/com/baidu/hugegraph/api/CypherApiTest.java:2]: +1

File: hugegraph-client/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java (hunk 3 above)
Comment (javeme) [reply hugegraph-client/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:2]: The original codes are also 2017, we can update them uniformly in the future.

**Review by imbajin — APPROVED** [review]

**Pull request merged**